### DISCUSSION TITLE
Issues with Incompatible language version.

### DISCUSSION THREAD

**Discussion by moontaeh** [tree-sitter/tree-sitter#1298] [Q&A] [answered]
I'm unsure if I'm doing something wrong here but I have written a personal grammar and generated it creating a parser.c file. However, I want to be able to using the node.js bindings for this parser but I whenever I generate the parser.c I get "LANGUAGE_VERSION 12".

However, when I try to run the parser.js file I created following the instructions from https://github.com/tree-sitter/node-tree-sitter I keep getting the error: 
"RangeError: Incompatible language version. Compatible range: 13 - 13. Got: 12"

Forcibly changing the language version to 12 seems to bypass the error but I then proceed to only get segmentation faults. I was wondering if there would be a way to generate the parser so that the language version ends up as 13.

I am currently using tree-sitter-cli v0.20.0 and tree-sitter v0.19.0 (I was unsure if the two were different things or not because they seemed to be different versions when I install them using npm).

**Comment by ahlinc** [comment, marked as answer]
Read my comment here: https://github.com/tree-sitter/tree-sitter-haskell/issues/35#issuecomment-857387920.
You need to check what versions of `tree-sitter-cli` exists on your machine and verify that only a one latest `tree-sitter-cli` `v0.20.x` version is used everywhere. First of all verify that there is no older version in your `node_modules` folder cause it will be used by the `node-gyp` when building your node.js binding.

**Reply by moontaeh** [reply]
Thank you so much for the help! :)
I had downloaded tree-sitter from various platforms due to initial installation problems but that seemed to have caused several versions to appear on my computer
Starting over from a clean slate with the installations as you stated fixed the problem!

**Comment by sogaiu** [comment]
Not quite sure I'm following but FWIW I have a setup where using multiple tree-sitter cli versions (0.19.3 and 0.20.0) is working for a single user account.

Currently, [emacs-tree-sitter doesn't appear to work with 0.20.0](https://github.com/emacs-tree-sitter/elisp-tree-sitter/commit/c95731dea6171f5d09b15bb7a2d0cd789ee65e71), apparently [due to a breaking change](https://github.com/tree-sitter/tree-sitter/pull/1157) so being able to use both is helpful.

At any rate, what seems to be working here for Linux is to have:

* both `~/.tree-sitter` (for 0.19.3) and `~/.config/tree-sitter` (for 0.20.0) directories
* specifying a single non-existent directory in `~/.tree-sitter/config.json` for the content of the `parser-directories` array:
    ```
      "parser-directories": [
        "/non-existent"
      ],
    ```
